_dev/sources/addin.ts

```diff
@@ -403,24 +450,50 @@ class Addin {
             rulesDependencies = this.rulesBuilder.getDependencies(this.data.rules);
             distributionListsDependencies = this.distributionListsBuilder.getDependencies(this.data.distributionLists);
             dependencies = this.combineDependencies(reportsDependencies, rulesDependencies, distributionListsDependencies);
+            this.tempUsers = results[6];
```

Comment:
Nick if you check a bit further on it's to provide the user the ability to add all users or zones:
            if(this.exportAllUsersCheckbox.checked==true){
                //sets exported users equal to all database users
                this.data.users = this.tempUsers;
            }
            if(this.exportAllZonesCheckbox.checked==true){
                //sets exported users equal to all database users
                this.data.zones = this.tempZones;
            }

Reply:
Whereas previously it only added dependent users & zones I think. These were change requests by the end users on our end to have the ability to export all or not (only dependent objects).

Reply:
Yes. But every User and Zone entity has its own dependencies so you must resolve them using `resolveDependencies` method. Because in some cases (e.g. when Zone is related to some ZoneType) you'll get an error during import process.

Reply:
Nick yes that process still happens I believe...with our update. Do you confirm?

Reply:
No. You replace Users and Zones on lines 459 - 470. But resolves all dependencies on line 455. So you don't have dependencies of Users and Zones entities into `this.data`.

Reply:
yes but if you have them all then all dependencies are taken care of

Reply:
But you have to take care of all dependencies. Otherwise your import can fail.

Reply:
I am not sure how to handle this. Example, If all users are selected and the user then de-selects this option I will need to render again is that correct? This to resolve the changed dependencies. Or are the dependencies resolved on the submit again? So on each check box change a render is executed. This is what I was trying to avoid with the Save button.

Reply:
ok so yes understood thanks. I will add before the resolve dependencies. thanks

Reply:
done.